client/src/app/_services/auth.service.ts

```diff
@@ -26,40 +26,30 @@ export class AuthService {
       if (currentUser.jwt) {
         const jwtDecoded = jwt_decode(currentUser.jwt);
         if (Date.now() < jwtDecoded.exp * 1000) {
-          this.managePermissionAndRole(jwtDecoded.permissions);
-
-          // Update /me
           this.me().subscribe();
-
-          // Refresh the token after 50ms to prevent other call.
-          setTimeout(
-            () => {
-              this.refresh().subscribe();
-            },
-            50,
-          );
+        } else {
+          this.clearUser();
+        }
+      }
+    }
+    if (sessionStorage.getItem('currentUser')) {
```

Comment:
Two branches are identical no? You could use the same system I said in interceptor. The only thing which might be different is the time to consider a jwt 'soon to be expired'.

You could add a field `longTime: true` in the localStorage object at login and test it here.